internal/uploading/file.go

```diff
@@ -0,0 +1,184 @@
+package uploading
+
+import (
+	"errors"
+	"time"
+)
+
+func NewFile(
+	uniqueId string,
+	name string,
+	clientExtension string,
+	extension string,
+	mimetype string,
+	dir string,
+	size uint32,
+	createdAt time.Time,
+	updatedAt time.Time,
+) (*File, error) {
+	file := &File{
+		uniqueId:        uniqueId,
+		name:            name,
+		clientExtension: clientExtension,
+		extension:       extension,
+		mimetype:        mimetype,
+		dirpath:         dir,
+		size:            size,
+		createdAt:       createdAt,
+		updatedAt:       updatedAt,
+	}
+
+	err := file.Validate()
+
+	return file, err
+}
+
+type File struct {
```

Comment:
is this necessary @yudium ?

Reply:
I see your point these code seem useless and we still need a place to put business logic especially the ones that shared across applications or "handlers" in our case. For example:

```go
func (o *File) GetFullpath() string {
	return o.dirpath + "/" + o.uniqueId
}
```

this can be used in delete and create handler. As you know, it usually called Entity.

How can we remove other unnecessary codes? If we look at it. Many of them just Getter and Setter and easily replaced by public properties:

```go
type File struct {
  UniqueId string; // no need more SetUniqueId() and GetUniqueId()
}
```

But I don't know this is OK because above example has mixed Data Structure and Object. But I found no issue until now to mix these.

Then should we use public data structure in entity to remove setter/getters?